common/persistence/serialization/codec.go

```diff
@@ -0,0 +1,68 @@
+package serialization
+
+import (
+	"errors"
+
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/server/common/codec"
+	"google.golang.org/protobuf/proto"
+)
+
+func ProtoEncode(m proto.Message) (*commonpb.DataBlob, error) {
+	return encodeBlob(m, enumspb.ENCODING_TYPE_PROTO3)
+}
+
+func encodeBlob(m proto.Message, encoding enumspb.EncodingType) (*commonpb.DataBlob, error) {
+	if m == nil {
```

Comment:
One of the helpers only tested for nil, the other one also tested via reflection. I tested it out and the reflection check made no difference (ie the marshaller returned `nil`). So it's not needed.